src/components/ProductCardGrid/ProductCardGrid.tsx

```diff
@@ -0,0 +1,37 @@
+import ProductCard from '../ProductCard/ProductCard';
+import { Product } from '../../types';
+import styled from 'styled-components';
+
+type Props = {
+  productList: Array<Partial<Product>>;
+};
+
+function ProductCardGrid({ productList }: Props) {
+  return (
+    <StyledProductCardGrid>
+      {productList.map((product) => (
+        <ProductCard product={product} key={product.id} />
+      ))}
+    </StyledProductCardGrid>
+  );
+}
+
+const StyledProductCardGrid = styled.div`
+  display: grid;
+  grid-gap: 40px;
+  width: max-content;
+
+  ${({ theme: { media } }) => media.sm`
+    grid-template-columns: repeat(1, minmax(0, 1fr));
+  `};
+
+  ${({ theme: { media } }) => media.md`
+    grid-template-columns: repeat(3, minmax(0, 1fr));
+  `};
+
+  ${({ theme: { media } }) => media.lg`
+    grid-template-columns: repeat(4, minmax(0, 1fr));
+  `};
+`;
```

Comment:
```suggestion
const StyledProductCardGrid = styled.div`
  display: grid;
  grid-gap: 40px;
  width: max-content;
  grid-template-columns: repeat(1, minmax(0, 1fr));
  ${({ theme: { media } }) => media.md`
    grid-template-columns: repeat(3, minmax(0, 1fr));
  `};

  ${({ theme: { media } }) => media.lg`
    grid-template-columns: repeat(4, minmax(0, 1fr));
  `};
`;
```

정하기 나름이지만 css작성할때 모바일이나 데스크탑 기준을 하나 정하면 위와같이 sm이나 lg 하나를 뺴고 작성할수도 있습니다. 참고정도로 해주세요